src/pykeen/models/base.py

```diff
@@ -490,6 +556,133 @@ def predict_scores_all_heads(
+
+        .. code-block:: python
+
+            from pykeen.pipeline import pipeline
+
+            # Train a model (quickly)
+            result = pipeline(model='RotatE', dataset='Nations', training_kwargs=dict(num_epochs=5))
+            model = result.model
+
+            # Get scores for *all* triples
+            tensor = model.score_all_triples()
+            df = model.make_labeled_df(tensor)
+
+            # Get scores for top 15 triples
+            top_tensor = model.score_all_triples(k=15)
+            top_df = model.make_labeled_df(tensor)
+        """
+        # set model to evaluation mode
+        self.eval()
+
+        # Do not track gradients
+        with torch.no_grad():
+            logger.warning(
+                f'score_all_triples is an expensive operation, involving {self.num_entities ** 2 * self.num_relations} '
+                f'score evaluations.'
+            )
+
+            if k is None:
+                logger.warning(
+                    'Not providing k to score_all_triples entails huge memory requirements for reasonably-sized '
+                    'knowledge graphs.'
+                )
+                return self._score_all_triples(batch_size=batch_size)
+
+            # initialize buffer on device
+            result = torch.ones(0, 3, dtype=torch.long, device=self.device)
+            scores = torch.empty(0, dtype=torch.float32, device=self.device)
+
+            for r, e in itt.product(
+                range(self.num_relations),
+                range(0, self.num_entities, batch_size),
+            ):
+                # calculate batch scores
+                hs = torch.arange(e, min(e + batch_size, self.num_entities), device=self.device)
+                hr_batch = torch.stack([
+                    hs,
+                    hs.new_empty(1).fill_(value=r).repeat(hs.shape[0])
+                ], dim=-1)
+                top_scores = self.predict_scores_all_tails(hr_batch=hr_batch)
+
+                # get top scores within batch
+                top_scores, top_indices = top_scores.view(-1).topk(k=k, largest=True, sorted=False)
+                top_heads, top_tails = top_indices // self.num_entities, top_indices % self.num_entities
+                top_triples = torch.stack([
+                    top_heads,
+                    top_heads.new_empty(top_heads.shape).fill_(value=r),
+                    top_tails,
+                ], dim=-1)
+
+                # append to global top scores
+                scores = torch.cat([scores, top_scores])
+                result = torch.cat([result, top_triples])
+
+                # reduce size if necessary
+                if result.shape[0] > k:
+                    scores, ind = scores.topk(k=k, largest=True, sorted=False)
+                    result = result[ind]
+
+            # Sort final result
+            ind = scores.argsort(descending=True)
+            result = result[ind]
+            # scores = scores[ind]
+
+            return result
```

Comment:
still need to add a keyword argument to include the scores as the fourth column of this tensor, or return a second 1-d tensor with the scores in order (either are fine for me)

Reply:
Done in a2e285c 

I did not use a keyword arguments, since just always returning `(result, scores)` does not need any extra code, and the performance/memory-wise there is not much to gain with having an extra code path based on knowing we do not care about the scores.

Reply:
do they need to be sorted by ind?